pinot-core/src/test/java/com/linkedin/pinot/server/realtime/ControllerLeaderLocatorTest.java

```diff
@@ -32,6 +32,69 @@
 
 public class ControllerLeaderLocatorTest {
 
+  /**
+   * Tests the invalidate logic for cached controller leader
+   * We set the value for lastCacheInvalidateMillis as we do not want to rely on operations being executed within or after the time thresholds in the tests
+   */
+  @Test
+  public void testInvalidateCachedControllerLeader()  {
+    HelixManager helixManager = mock(HelixManager.class);
+    HelixDataAccessor helixDataAccessor = mock(HelixDataAccessor.class);
+    BaseDataAccessor<ZNRecord> baseDataAccessor = mock(BaseDataAccessor.class);
+    ZNRecord znRecord = mock(ZNRecord.class);
+    final String leaderHost = "host";
+    final int leaderPort = 12345;
+
+    when(helixManager.getHelixDataAccessor()).thenReturn(helixDataAccessor);
+    when(helixDataAccessor.getBaseDataAccessor()).thenReturn(baseDataAccessor);
+    when(znRecord.getId()).thenReturn(leaderHost + "_" + leaderPort);
+    when(baseDataAccessor.get(anyString(), any(), anyInt())).thenReturn(znRecord);
+    when(helixManager.getClusterName()).thenReturn("testCluster");
+
+    // Create Controller Leader Locator
+    FakeControllerLeaderLocator.create(helixManager);
+    ControllerLeaderLocator controllerLeaderLocator = FakeControllerLeaderLocator.getInstance();
+
+    // check values at startup
+    Assert.assertTrue(controllerLeaderLocator.isCachedControllerLeaderInvalid());
+    Assert.assertEquals(controllerLeaderLocator.getLastCacheInvalidateMillis(), 0);
+
+    // very first invalidate
+    controllerLeaderLocator.invalidateCachedControllerLeader();
+    Assert.assertTrue(controllerLeaderLocator.isCachedControllerLeaderInvalid());
+    long lastCacheInvalidateMillis = controllerLeaderLocator.getLastCacheInvalidateMillis();
+    Assert.assertTrue(lastCacheInvalidateMillis > 0);
```

Comment:
a better test will be to take the millis before calling invaidate, and assert that the invalidation time is >= that time stamp,

Reply:
that is being done for one of the following tests, where we want to test for lastCacheInvalidMillis being updated wrt the older value. For this particular one, I wanted to test that initial value has been updated from 0